### DISCUSSION TITLE
Assistants v2 and Vector Store API support

### DISCUSSION THREAD

**Discussion by gehrisandro** [openai-php/client#411] [Announcements]
Support for the Assistants API v2 and the Vector Stores API as finally arrived (beta). 🚀

Happy to hear your feedback before making it a stable release.

https://github.com/openai-php/client/releases/tag/v0.10.0-beta.1

Thanks to @knash94 for his valuable help on this one!

**Comment by neverything** [comment]
Thank you @gehrisandro! I just tested streaming using `createAndRunStreamed` in a quick and dirty Laravel Livewire component using an assistant that I feed lots of conference data. Quick and dirty, but works very well:

```php
public function getAssistantResponse()
{
    $user_question = collect($this->messages)->where('role', 'user')->toArray();

    $stream = app('openai')->threads()->createAndRunStreamed([
        'assistant_id' => 'asst_XXXXXXXXXXXXXXXX',
        'thread' => [
            'messages' => $user_question,
        ],
    ]);

    foreach ($stream as $response) {
        if ($response->event === 'thread.message.delta') {
            $content = Arr::get($response->response->delta->content[0]->toArray(), 'text.value');

            $this->response .= $content;

            $this->stream(
                to: 'stream-'.$this->getId(),
                content: $content,
                replace: false
            );
        }

        if ($response->event === 'thread.message.completed') {
            $content = Arr::get($response->response->content[0]->toArray(), 'text.value');

            $this->response = $content;

            $this->stream(
                to: 'stream-'.$this->getId(),
                content: Str::markdown($content),
                replace: true
            );
        }
    }
}
```

**Reply by gehrisandro** [reply]
Thank you @neverything for your Feedback 🎉

**Comment by knash94** [comment]
I've just switched to the beta tag release and everything is working as it was on my branch. Thanks for getting this into the main repo @gehrisandro 🎉

**Reply by gehrisandro** [reply]
@knash94 Thanks for your feedback!

**Comment by kieranx** [comment]
New API update on this today with some new features - @gehrisandro can we get these in? https://twitter.com/OpenAIDevs/status/1798037560905855186

**Reply by gehrisandro** [reply]
Hi @kieranx 

Thanks for the hint. Added here: https://github.com/openai-php/client/pull/419

**Reply by kieranx** [reply]
Hey @gehrisandro - did you miss this part, I don’t see it listed in the examples? 

Number of chunks

By default, the file_search tool outputs up to 20 chunks for gpt-4* models and up to 5 chunks for gpt-3.5-turbo. You can adjust this by setting [file_search.max_num_results](https://platform.openai.com/docs/api-reference/assistants/createAssistant#assistants-createassistant-tools) in the tool when creating the assistant or the run.

The file_search.max_num_results is also new.

**Reply by gehrisandro** [reply]
You are right, I missed this one.

Could you make a PR?

**Reply by kieranx** [reply]
I'm not a dev myself, but we use this package in our SaaS and I'm watching this functionality closely... unfortunately I have no idea how to make the PR etc - sorry!

**Reply by gehrisandro** [reply]
No problem, I can do it, but not right now, as I am busy and this is just a side project of mine.

In the meantime, you may consider [supporting me](https://github.com/sponsors/gehrisandro) 🙏🏼

**Comment by gehrisandro** [comment]
Support is here: https://github.com/openai-php/client/releases/tag/v0.10.1 🚀

**Comment by kieranx** [comment]
I just dropped you some support :-) appreciate your work on the project.
I'll keep an eye out for this update, as it's one we're waiting for. Cheers.

**Reply by gehrisandro** [reply]
Thanks a lot 🎉
